data/org.freedesktop.portal.Notification.xml

```diff
@@ -131,14 +387,43 @@
       <arg type="s" name="id" direction="in"/>
     </method>
 
+    <!--
+        SupportedOptions:
+
+        Some properties in :ref:`org.freedesktop.portal.Notification.AddNotification`
+        may have options advertised by the server.
+
+        Currently there are the following options the notification server can advertise:
+
+        * ``category`` (``as``)
+
+          Categories that the notification server understands and supports.
+
+        * ``button-purpose`` (``as``)
+
+          Button purposes that the notification server understands and supports.
+          This is the list of options that can be used as `purpose` for `buttons`.
+    -->
+    <property name="SupportedOptions" type="a{sv}" access="read"/>
+
     <!--
         ActionInvoked:
```

Comment:
Because this is modeled after `org.freedesktop.Application.ActivateAction`. Shouldn't this also split out the platform_data into its own argument instead of sending it in the @parameter array?

Reply:
would be nice, but doesn't that break backwards compatibility?

Reply:
I think a second signal would be better. Adding a new value into `parameter` may count as an API break anyway, because user code may be strictly validating the `parameter` they receive, and not expect an additional entry in the `av`.

Reply:
why would that count as an API break? It's defined as a `av` already.

Reply:
Hmm, does this `parameter` not match the `parameter` of a `GAction`? Or does it just look similar but have different semantics? Reading the rest of the spec it looks like `parameter` is actually just a dumping ground for vaguely-typed response data? Typically an `a{sv}` with well-defined-but-extensible keys would be used for this in a D-Bus API, but changing to that would definitely be an API break (or require a new signal).

Reply:
Having two similar but slightly different interfaces will be really confusing. Having a `ActionInvoked2` which matches the arguments from `org.freedesktop.Application.ActivateAction` would be an improvement (and we could add a `a{sv}` just for good measure).

Reply:
> Having two similar but slightly different interfaces will be really confusing.

You mean `org.freedesktop.Application.ActivateAction` and the `ActionInvoked` signal? 

The two different interfaces are pretty much never used in the same application. I think it may be a little confusing (but already having the two is confusing enough), and requiring applications, or client libs, to implement a third interface may be even more confusing, since they need to support version one 1 the interface.

I agree it's sub optimal but i strongly think that adding a `ActionInvoked2` creates unnecessary extra work for applications (or client libraries).

Reply:
I still really don't like this. The upside is that we could still add the `ActionInvoked2` signal later on. @pwithnall any more thoughts?

Reply:
I feel like it may make sense to just deprecate the signal and require applications to switch to using D-Bus Activation. Let's see what KDE people think @nicolasfella